### ISSUE TITLE
Marker return in a previously position if another marker is pressed during an animations

### ISSUE THREAD

**Issue by carlodenardin** [rorystephenson/flutter_map_marker_popup#39]
I have an issue with the animation of the markers. I setted the `MarkerCenterAnimation` but when I press a marker, if during this animation I press another marker the new marker pressed is unreachable and the map is going crazy.

This is an example:
https://imgur.com/a/RjqVud0

I setted an high duration time of the animation to mark the problem.

I didn't retrieve any error in the console.

This is my `flutter doctor`:
```
[✓] Flutter (Channel master, 2.11.0-0.0.pre.752, on macOS 12.2.1 21D62 darwin-arm, locale en-GB)
    • Flutter version 2.11.0-0.0.pre.752 at /Users/carlodenardin/Workspace/Dependencies/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a786f65700 (32 hours ago), 2022-03-03 03:01:22 -0500
    • Engine revision 993285f64d
    • Dart version 2.17.0 (build 2.17.0-165.0.dev)
    • DevTools version 2.11.1

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/carlodenardin/Library/Android/sdk
    • Platform android-31, build-tools 30.0.3
    • Java binary at: /Library/Java/JavaVirtualMachines/jdk-11.0.11.jdk/Contents/Home/bin/java
    • Java version Java(TM) SE Runtime Environment 18.9 (build 11.0.11+9-LTS-194)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Users/carlodenardin/Downloads/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] VS Code (version 1.65.0)
    • VS Code at /Users/carlodenardin/Workspace/Ide/Visual Studio Code.app/Contents
    • Flutter extension version 3.36.0

[✓] Connected device (2 available)
    • iPhone 13 (mobile) • 0B09936E-70C6-4B29-8E70-594917B29350 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome
      99.0.4844.51

[✓] HTTP Host Availability
    • All required HTTP hosts are available
```

**Comment by rorystephenson** [comment]
Hi @carlodenardin .

That definitely doesn't look good! I'm going to try to figure out what's going on.

**Comment by rorystephenson** [comment]
@carlodenardin I've written a fix which is part of the new version (2.1.2). Please let me know if this is still an issue!

Thank you for a very detailed bug report and even the gif demonstrating the problem. That makes it so much easier to resolve!

**Comment by carlodenardin** [comment]
@rorystephenson I update the package to the new version and now the problem has been fixed. Thanks!
